src/server/pkg/testutil/auth.go

```diff
@@ -0,0 +1,364 @@
+package testutil
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/pachyderm/pachyderm/src/client"
+	"github.com/pachyderm/pachyderm/src/client/auth"
+	"github.com/pachyderm/pachyderm/src/client/pkg/errors"
+	"github.com/pachyderm/pachyderm/src/client/pkg/require"
+	"github.com/pachyderm/pachyderm/src/server/pkg/backoff"
+)
+
+const (
+	// AdminUser is the sole cluster admin after GetAuthenticatedPachClient is called
+	AdminUser = auth.RobotPrefix + "admin"
+
+	adminTokenFile = "/tmp/pach-auth-test_admin-token"
+)
+
+var (
+	tokenMapMut sync.Mutex // guards both tokenMap and seedClient
+	tokenMap    = make(map[string]string)
+	seedClient  *client.APIClient
+)
+
+// helper function that prepends auth.GitHubPrefix to 'user'--useful for validating
+// responses
+func gh(user string) string {
+	return auth.GitHubPrefix + user
+}
+
+// helper method to generate a super admin role
+func superClusterRole() *auth.ClusterRoles {
+	return &auth.ClusterRoles{Roles: []auth.ClusterRole{auth.ClusterRole_SUPER}}
+}
+
+// UpdateAuthToken manually updates the auth token for a given user
+func UpdateAuthToken(user string, token string) {
```

Comment:
A few activation tests involve minting new admin tokens - this provides a way to update the cache so `DeleteAll` still succeeds in the test teardown.